### PR TITLE
[12.0][IMP] - different receivable/payable accounts by brand

### PR DESCRIPTION
**Pull request by sbejaoui**
This PR add a new model to set different receivable/payable accounts for partners by brand.

### CODE REVIEW

**Comment by sbejaoui** [comment]
FYI @sbidoul , @gva-acsone

**Comment by sbejaoui** [comment]
@max3903 can you review this please

File: account_brand/models/res_partner_account_brand.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+class ResPartnerAccountBrand(models.Model):
+    """This model is meant to be used in case we need to define different
+    receivable/payable accounts for partners"""
+
+    _name = "res.partner.account.brand"
+    _description = "Receivable/Payable Partner Account By Brand"
+
+    partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Partner", required=False
```
Comment (nikul-serpentcs) [inline account_brand/models/res_partner_account_brand.py:15]: @sbejaoui No need to `required=False`

File: account_brand/models/res_partner_account_brand.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    )
+    account_id = fields.Many2one(
+        comodel_name="account.account", string="Account", required=True
```
Comment (nikul-serpentcs) [inline account_brand/models/res_partner_account_brand.py:18]: same here

File: account_brand/models/account_invoice.py [hunk 3]
```diff
@@ -1,11 +1,52 @@
...
+                partner_account_brand_model._get_partner_account_by_brand(
+                    "payable", self.brand_id, partner
+                )
+            )
+            pay_account = (
+                pay_account.account_id
+                if pay_account
+                else partner.property_account_payable_id
+            )
+            if invoice_type in ("in_invoice", "in_refund"):
```
Comment (nikul-serpentcs) [inline account_brand/models/account_invoice.py:47]: @sbejaoui I think here no need to `else`
```
account_id = rec_account
if invoice_type in ("in_invoice", "in_refund"):
    account_id = pay_account

```

**Review by nikul-serpentcs — COMMENTED** [review]
Minor Change

**Comment by sbejaoui** [comment]
This PR was moved to [Brand\#5](https://github.com/OCA/brand/pull/5)

**Pull request closed without merging**